tasks/setup-debian-vanilla.yml

```diff
@@ -1,37 +1,27 @@
 ---
-- name: (Debian) Install GPG - required to add wireguard key
-  apt:
-    name: gnupg
-    state: present
-
-- name: (Debian) Add WireGuard repository on buster or earlier
+- name: (Debian) Add WireGuard repository on buster
   apt_repository:
     repo: "deb http://deb.debian.org/debian buster-backports main"
-    state: present
+    state: "{{ 'present' if (ansible_distribution_version | int <= 10) else 'absent' }}"
     update_cache: yes
-  when: ansible_distribution_version | int <= 10
   tags:
     - wg-install
 
 - name: (Debian) Get architecture
   command: "dpkg --print-architecture"
-  register: dpkg_arch
+  register: wireguard__fact_dpkg_arch
   changed_when: False
 
-- set_fact:
-    kernel_header_version: "{{ ('-cloud-' in ansible_kernel) | ternary(ansible_kernel,dpkg_arch.stdout) }}"
-
 - name: (Debian) Install kernel headers to compile Wireguard with DKMS
   apt:
     name:
-      - "linux-headers-{{ kernel_header_version }}"
+      - "linux-headers-{{ ('-cloud-' in ansible_kernel) | ternary(ansible_kernel,wireguard__fact_dpkg_arch.stdout) }}"
     state: present
 
-- name: (Debian) Install wireguard packages
+- name: (Debian) Install WireGuard packages
   apt:
     name:
-      - "wireguard-dkms"
-      - "wireguard-tools"
```

Comment:
What happens if the two packages are already present? Are they conflicting with `wireguard` package or will this just work because `wireguard` is a meta package?

Reply:
It is a small package with:

```
Depends: wireguard-modules (>= 0.0.20191219) | wireguard-dkms (>= 0.0.20200121-2), wireguard-tools (>= 1.0.20200827-1~bpo10+1)
```

So no conflict. The added benefit is what is written in `/usr/share/doc/wireguard/TODO.Debian`:

```
* wireguard metapackage should accept a 5.6 (or later) kernel instead
  of a module
```

So they will just update the `Depends:` line to accept a recent enough kernel.